V2_MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,183 @@
+# V2 Migration Guide
+
+## Configuration
+
+### Required Keys
+
+- `appSession.secret` is now just `secret` (because it is now used to sign the transient cookies as well as the `appSession` cookie). The environmental variable has changed from `APP_SESSION_SECRET` to `SECRET`.
+
+```dotenv
+# Before
+SESSION_SECRET=LONG_RANDOM_STRING
+
+# After
+SECRET=LONG_RANDOM_STRING
+```
+
+```js
+const { auth } = require('express-openid-connect');
+
+// Before
+app.use(
+  auth({
+    session: {
+      secret: 'LONG_RANDOM_STRING',
+    },
+  })
+);
+
+// After
+app.use(
+  auth({
+    secret: 'LONG_RANDOM_STRING',
+  })
+);
+```
+
+### Route customization
+
+You can now enable individual routes rather than all or nothing and the path to routes can be customized using the `routes` config.
+
+```js
+// Before
+app.use(
+  auth({
+    routes: true,
+    loginPath: '/custom/login',
+    logoutPath: '/custom/logout',
+    redirectUriPath: '/custom/callback',
+    postLogoutRedirectUri: '/custom/post-logout',
+  })
+);
+
+// After
+app.use(
+  auth({
+    routes: {
+      login: '/custom/login',
+      logout: '/custom/logout',
+      callback: '/custom/callback',
+      postLogoutRedirect: '/custom/post-logout',
+    },
+  })
+);
+```
+
+### Session Lifecycle configuration
+
+Session duration was rolling only and configured using `appSession.duration` (default 24hrs). Now it can be configured to rolling or absolute (default rolling 24hrs and absolute 7 days).
+
+```js
+// Before
+app.use(
+  auth({
+    appSession: {
+      duration: 86400, // default 1 day in secs
+    },
+  })
+);
+
+// After
+app.use(
+  auth({
+    appSession: {
+      rolling: true,
+      rollingDuration: 86400, // default 1 day rolling duration in secs
+      absoluteDuration: 86400 * 7, // default 7 days absolute duration in secs
+    },
+  })
+);
+```
+
+### `required` is now `authRequired`
+
+To enable or disable all routes to require authentication, use the `authRequired` configuration (default `true`)
+
+```js
+// Before
+app.use(
+  auth({
+    required: true,
+  })
+);
+
+// After
+app.use(
+  auth({
+    authRequired: true,
+  })
+);
+```
+
+### Configuration items renamed/removed
+
+- **`required` is now `authRequired`** - to enable or disable all routes to require authentication, use the `authRequired` configuration (default `true`)
+- **`idTokenAlg` is now `idTokenSigningAlg`** - to specify an id token signing algorithm, use `idTokenSigningAlg`
+- **`httpOptions`** - it is no longer possible to pass custom http options to the underlying library
```

Comment:
```suggestion
- **`httpOptions`** was removed - it is no longer possible to pass custom http request options to the underlying library. It will be again in the future in a more curated and comprehensive feature we have in mind for this library.
```